src/index.js

```diff
@@ -0,0 +1,17 @@
+const { IPinfoWrapper } = require("node-ipinfo");
+const defaultIPSelector = require("./ip-selector/default-ip-selector");
+
+module.exports = ({ token = "", cache, timeout, ipSelector }) => {
+    const ipinfo = new IPinfoWrapper(token, cache, timeout);
+    if (ipSelector == null || typeof ipSelector != "function") {
+        ipSelector = defaultIPSelector;
+    }
+    return async (req, _, next) => {
+        const ip = ipSelector(req);
+        req.ipinfo = await ipinfo.lookupIp(ip);
+        next();
+    };
+};
+
+module.exports.defaultIPSelector = defaultIPSelector;
+module.exports.originatingIPSelector = require("./ip-selector/originating-ip-selector");
```

Comment:
For consistency let's put this require at the top. (It does pollute the namespace a bit but it's not like we're gonna hit a conflict).